tests/test_async.py

```diff
@@ -0,0 +1,52 @@
+import uuid
```

Comment:
I think we need a LOT more async tests than this. In particular, we need tests exercising every part of the async API just like for the sync API--because there's so much code duplication we need to test both sides of the code.

Depending on our API, we may also need tests for workflows that mix async and sync steps. If we don't want to support those, we have to deliberately design our API to make that impossible, for example by having a `DBOS` and `AsyncDBOS`.

Reply:
Agreed, though I'm not sure separating Sync vs Async DBOS will make it *impossible*